sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py

```diff
@@ -46,8 +46,11 @@
 async def process_content(data: Any, start_offset: int, end_offset: int, encryption: Dict[str, Any]) -> bytes:
     if data is None:
         raise ValueError("Response cannot be None.")
-    await data.response.load_body()
-    content = cast(bytes, data.response.body())
+    if not data.response.is_stream_consumed:
+        content = b"".join([d async for d in data])
```

Comment:
This is a good start-- then test against the 3 transports, then when those work test `validate_content`